store/store.go

```diff
@@ -131,6 +131,24 @@ func infoFromRemote(d *snapDetails) *snap.Info {
 		info.Contact = d.SupportURL
 	}
 
+	// fill in the plug/slot data
+	if rawYamlInfo, err := snap.InfoFromSnapYaml([]byte(d.SnapYAML)); err == nil {
+		if info.Plugs == nil {
+			info.Plugs = make(map[string]*snap.PlugInfo)
+		}
+		for k, v := range rawYamlInfo.Plugs {
+			info.Plugs[k] = v
+			info.Plugs[k].Snap = info
+		}
+		if info.Slots == nil {
+			info.Slots = make(map[string]*snap.SlotInfo)
+		}
+		for k, v := range rawYamlInfo.Slots {
+			info.Slots[k] = v
+			info.Slots[k].Snap = info
+		}
+	}
```

Comment:
This looks better but I wonder if you need look into `info.{Plugs,Slots}.{Apps,Hooks}`?

Reply:
this is done via `snap.InfoFromSnapYaml([]byte(d.SnapYAML)` automatically AFAICT. I added an extra test for this now just to be sure.